plugins/inputs/ceph/ceph.go

```diff
@@ -411,45 +441,96 @@ func decodeStatus(acc telegraf.Accumulator, input string) error {
 	return nil
 }
 
+// decodeStatusFsmap decodes the FS map portion of the output of 'ceph -s'
+func decodeStatusFsmap(acc telegraf.Accumulator, data *CephStatus) error {
+	fields := map[string]interface{}{
+		"in":         data.FSMap.NumIn,
+		"max":        data.FSMap.NumMax,
+		"up_standby": data.FSMap.NumUpStandby,
+		"up":         data.FSMap.NumUp,
+	}
+	acc.AddFields("ceph_fsmap", fields, map[string]string{})
+	return nil
+}
+
 // decodeStatusHealth decodes the health portion of the output of 'ceph status'
 func decodeStatusHealth(acc telegraf.Accumulator, data *CephStatus) error {
+	statusCodes := map[string]float64{
+		"HEALTH_ERR":  0,
+		"HEALTH_WARN": 1,
+		"HEALTH_OK":   2,
+	}
 	fields := map[string]interface{}{
+		"overall_status": data.Health.OverallStatus, // This field is no longer reported in ceph 10 and later
+		"status_code":    statusCodes[data.Health.Status],
 		"status":         data.Health.Status,
-		"overall_status": data.Health.OverallStatus,
 	}
 	acc.AddFields("ceph_health", fields, map[string]string{})
 	return nil
 }
 
+// decodeStatusMonmap decodes the Mon map portion of the output of 'ceph -s'
+func decodeStatusMonmap(acc telegraf.Accumulator, data *CephStatus) error {
+	fields := map[string]interface{}{
+		"num_mons": data.MonMap.NumMons,
+	}
+	acc.AddFields("ceph_monmap", fields, map[string]string{})
+	return nil
+}
+
 // decodeStatusOsdmap decodes the OSD map portion of the output of 'ceph -s'
 func decodeStatusOsdmap(acc telegraf.Accumulator, data *CephStatus) error {
 	fields := map[string]interface{}{
-		"epoch":            data.OSDMap.OSDMap.Epoch,
-		"num_osds":         data.OSDMap.OSDMap.NumOSDs,
-		"num_up_osds":      data.OSDMap.OSDMap.NumUpOSDs,
-		"num_in_osds":      data.OSDMap.OSDMap.NumInOSDs,
-		"full":             data.OSDMap.OSDMap.Full,
-		"nearfull":         data.OSDMap.OSDMap.NearFull,
-		"num_remapped_pgs": data.OSDMap.OSDMap.NumRemappedPGs,
+		"epoch":            data.OSDMap.Epoch,
+		"full":             data.OSDMap.Full,
+		"nearfull":         data.OSDMap.NearFull,
+		"num_in_osds":      data.OSDMap.NumInOSDs,
+		"num_osds":         data.OSDMap.NumOSDs,
+		"num_remapped_pgs": data.OSDMap.NumRemappedPGs,
+		"num_up_osds":      data.OSDMap.NumUpOSDs,
+	}
+	if data.OSDMap.Epoch == 0 && data.OSDMap.NumOSDs == 0 {
+		fields = map[string]interface{}{
```

Comment:
What about testing if `data.OSDMap.OSDMap` is set? That seems more reliable to me.

Reply:
Very true, latest update makes that change.

Reply:
You are sure `data.OSDMap.OSDMap.NumOSDs` will never be `0`?